src/libraries/System.Net.Sockets/tests/FunctionalTests/SendReceive/SendReceive.cs

```diff
@@ -1228,6 +1227,40 @@ public sealed class SendReceive_MemoryArrayTask : SendReceive<SocketHelperMemory
     {
         public SendReceive_MemoryArrayTask(ITestOutputHelper output) : base(output) { }
 
+        [Fact]
+        public async Task Send_0ByteSend_Memory_Success()
```

Comment:
Maybe make this a theory and test out different inputs, e.g.
- `ReadOnlyMemory<byte>.Empty`
- `(ReadOnlyMemory<byte>)Array.Empty<byte>()`
- `new ReadOnlyMemory<byte>(new byte[1], 0, 0)`

etc.

Reply:
The non-memory-specific version of this will cover your second example. 

I'm not sure there's enough variation in the third example to make it worth adding?

Reply:
I'm fine either way.